relay-server/src/services/processor.rs

```diff
@@ -2501,12 +2501,11 @@ impl EnvelopeProcessorService {
 
         let CardinalityLimitsSplit { accepted, rejected } = limits.into_split();
 
-        if !rejected.is_empty() {
+        for (bucket, exceeded) in rejected {
```

Comment:
We could group buckets by the limit they exceeded and track them in bulk, but I'm not sure it's worth the complication.